targets/simple_switch/primitives.cpp

```diff
@@ -274,47 +280,51 @@ class copy_header : public ActionPrimitive<Header &, const Header &> {
 
 REGISTER_PRIMITIVE(copy_header);
 
-/* standard_metadata.clone_spec will contain the mirror id (16 LSB) and the
-   field list id to copy (16 MSB) */
 class clone_ingress_pkt_to_egress
   : public ActionPrimitive<const Data &, const Data &> {
-  void operator ()(const Data &clone_spec, const Data &field_list_id) {
-    Field &f_clone_spec = get_field("standard_metadata.clone_spec");
-    f_clone_spec.shift_left(field_list_id, 16);
-    f_clone_spec.add(f_clone_spec, clone_spec);
+  void operator ()(const Data &mirror_session_id, const Data &field_list_id) {
+    auto &packet = get_packet();
+    // Assume mirror_session_id < (1u << 15) so that we can use most
```

Comment:
Could we enforce this when the control-plane creates a session here: https://github.com/p4lang/behavioral-model/blob/master/targets/simple_switch/simple_switch.cpp#L78, i.e. log an error and return `false` is the session id is `>= (1 << 15)`?

Reply:
updated this